mmdet3d/datasets/dataset_wrappers.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+from .builder import DATASETS
+
+
+@DATASETS.register_module()
+class CBGSDataset(object):
+    """A wrapper of class sampled dataset with ann_file path. Implementation of
+    paper `Class-balanced Grouping and Sampling for Point Cloud 3D Object
+    Detection <https://arxiv.org/abs/1908.09492.>`_.
+
+    Balance the number of scenes under different classes.
+
+    Args:
+        dataset (:obj:`CustomDataset`): The dataset to be class sampled.
+    """
+
+    def __init__(self, dataset):
+        self.dataset = dataset
+        self.CLASSES = dataset.CLASSES
+        self.sample_indices = self._get_sample_indices()
+        # self.dataset.data_infos = self.data_infos
+        if hasattr(self.dataset, 'flag'):
+            self.flag = np.array(
+                [self.dataset.flag[ind] for ind in self.sample_indices],
+                dtype=np.uint8)
+
+    def _get_sample_indices(self):
+        """Load annotations from ann_file.
+
+        Args:
+            ann_file (str): Path of the annotation file.
+
+        Returns:
+            list[dict]: List of annotations after class sampling.
+        """
+        data = self.dataset.data_infos
+        class_sample_idx = {name: [] for name in self.CLASSES}
+        for idx, info in enumerate(data):
+            if self.dataset.use_valid_flag:
```

Comment:
1. For line 40 - 47, we may abstract the logics to implement a new function `sefl.dataset.get_cat_ids()` for each dataset, thus any existing dataset could use `CBGSDataset`.
2. Add unittest for `CBGSDataset`.